File: manage-company/src/main/front/src/app/models/api-error.ts
import {plainToClass} from 'class-transformer';

export function tryCastAndThrowApiError(error: any) {
    const customError = error.error;
    if (customError && customError.code) {
        throw plainToClass(ApiError, customError);
    }
}

export enum ApiErrorCode {
    NotAuthenticated = 100,
    Unauthorized = 110,
    InvalidAccountNameOrPassword = 115,
    LoginFailureBlock = 117,
    AccountNotActive = 120,
    InvalidParameters = 200
}

export class ApiError {
    constructor(public code: ApiErrorCode, public message: string) {
    }
}

File: manage-company/src/main/java/pl/managecompany/api/ApiError.java
package pl.managecompany.api;

import lombok.Data;

@Data
public class ApiError {
    private int code;
    private String message;

    public ApiError(ApiErrorCode apiErrorCode) {
        super();
        this.code = apiErrorCode.codeNumber;
    }

    public ApiError(ApiErrorCode apiErrorCode, String message) {
        super();
        this.code = apiErrorCode.codeNumber;
        this.message = message;
    }
}
